source/configuring-ember/preparing-for-upgrade.md

```diff
@@ -0,0 +1,154 @@
+    { handler: "silence", matchMessage: "In Ember 2.0 service factories must have an `isServiceFactory` property set to true. You registered (unknown mixin) as a service factory. Either add the `isServiceFactory` property to this factory or extend from Ember.Service." },
+    { handler: "silence", matchMessage: "Ember.View is deprecated. Consult the Deprecations Guide for a migration strategy." },
+    { handler: "silence", matchMessage: "Depending on arrays using a dependent key ending with `@each` is deprecated. Please refactor from `Ember.observer('content.@each', function() {});` to `Ember.computed('content.[]', function() {})`." },
+    { handler: "silence", matchMessage: /Accessing 'template' in .* is deprecated. To determine if a block was specified to .* please use '{{#if hasBlock}}' in the components layout./ },
+    { handler: "silence", matchMessage: "A property of <web-directory@view:-outlet::ember1012> was modified inside the didInsertElement hook. You should never change properties on components, services or models during didInsertElement because it causes significant performance degradation." }
+  ]
+};
+```
+
+![Failed Testcase from a Thrown Deprecation](../../images/guides/configuring-ember/preparing-for-upgrade/failed-test-from-deprecation.png)
+
+
+### 3. Fix and Repeat
+After fixing a deprecation and getting your scenarios working again, you might want to leave the deprecation message in the workflow file with the
+throw handler enabled.  This will ensure you haven't missed anything, and ensure no new deprecated calls of that type are introduced to your project.
+Next its just a matter of going down the list, changing the handler, and fixing.
+
+
+In the end your deprecations can be fully turned on as "throw" and you should be able to use your application without error.  At this point you can
+go ahead and up your ember version!  When you upgrade, be sure you remove the deprecation list you've fixed, and update your deprecation handler
+appropriately so that you can begin preparing for the next release.
+
+## Silencing Deprecation Warnings During Compile
+
+You might also notice as you upgrade between releases, that your terminal log begins to stream tons of deprecation warnings during your compile process, making
+it difficult to review your compilation logs.
+
+<img height="800px" src="../../images/guides/configuring-ember/preparing-for-upgrade/compile-deprecations.png" title="Compile Deprecations Clouding Log"/>
+
+If you are using the deprecation workflow process above you will likely prefer to gather these same warnings during runtime execution.  The way to hide these
+warnings during compile is to install the [ember-cli-template-lint](http://emberobserver.com/addons/ember-cli-template-lint) addon.  It supresses
+deprecation warnings during compile in favor of showing them in the browser console during test suite execution.
```

Comment:
It only shows them during the test suite? What about when the app itself is being used?